src/helpers.ts

```diff
@@ -73,6 +83,39 @@ export function generateBaseRouter(sourceFile: SourceFile, config: Config) {
     }`);
 }
 
+export function generateBaseRouterv10(sourceFile: SourceFile, config: Config) {
+  sourceFile.addStatements(/* ts */ `
+  import { Context } from '${config.contextPath}';
+  `);
+
+  sourceFile.addStatements(/* ts */ `
+  export const t = trpc.initTRPC.context<Context>().create();
+  `);
+
+  sourceFile.addStatements(/* ts */ `
+  export const publicProcedure = t.procedure;`);
+
+  const middlewares = [];
+  if (config.withMiddleware) {
+    middlewares.push(/* ts */ `
+    .use(({ ctx, next }) => {
+      console.log("inside middleware!")
+      return next();
+    })`);
+  }
```

Comment:
Here's how the shield will be used:

```ts
export const permissionsMiddleware = t.middleware(permissions)

export const shieldedProcedure = t.procedure.use(permissionsMiddleware)

export const publicProcedure = t.procedure

```

Then inside routers:

```ts
import { shieldedProcedure, t } from './helpers/createRouter'

export const usersRouter = t.router({
  createOneUser: shieldedProcedure.input(UserCreateOneSchema).mutation(async ({ ctx, input }) => {
    const createOneUser = await ctx.prisma.user.create(input)
    return createOneUser
  }),
});
```

If with middleware:

```ts
export const shieldedProcedure = t.procedure.use(globalMiddleware).use(permissionsMiddleware)

```